sway/commands/output/toggle.c

```diff
@@ -0,0 +1,37 @@
+#include "sway/commands.h"
+#include "sway/config.h"
+#include "sway/output.h"
+
+struct cmd_results *output_cmd_toggle(int argc, char **argv) {
+	if (!config->handler_context.output_config) {
+		return cmd_results_new(CMD_FAILURE, "Missing output config");
+	}
+
+	struct output_config *oc = config->handler_context.output_config;
+
+	if (strcmp(oc->name, "*") == 0) {
+		return cmd_results_new(CMD_INVALID,
+				"Cannot apply toggle to all outputs.");
+	}
+
+	struct sway_output *sway_output = all_output_by_name_or_id(oc->name);
+
+	if (sway_output == NULL) {
+		return cmd_results_new(CMD_FAILURE,
+				"Cannot apply toggle to unknown output %s", oc->name);
+	}
+
+	oc = find_output_config(sway_output);
```

Comment:
`oc` may be `NULL`. I'd suggest just flipping the if statement below and doing `if (!oc || oc->enabled == 1)`

Reply:
at the beginning `oc->enabled` is -1, as you suggested, so the first toggle would fail.

I can do `(!oc || oc->enabled == 1 || oc->enabled == -1)`?

Reply:
Just use `!oc || oc->enabled != 0`